packages/vite/src/node/optimizer/registerMissing.ts

```diff
@@ -267,14 +274,14 @@ export function createOptimizedDeps(server: ViteDevServer): OptimizedDeps {
         } else {
           commitProcessing()
 
-          if (newDepsToLogHandle) clearTimeout(newDepsToLogHandle)
-          newDepsToLogHandle = undefined
-          logNewDeps()
+          if (!isDebugEnabled) {
+            if (newDepsToLogHandle) clearTimeout(newDepsToLogHandle)
+            newDepsToLogHandle = undefined
+            logNewlyDiscoveredDeps()
+          }
 
           logger.info(
-            colors.green(
-              `✨ previous optimized dependencies have changed, reloading page`
-            ),
+            colors.green(`✨ optimized dependencies changed. reloading`),
```

Comment:
Maybe we could use a different emoji for this one. Like 🔄 or ⚙️. Not convinced by other options, probably better to keep the current one